pkg/runtime/interfaces.go

```diff
@@ -16,6 +16,24 @@ limitations under the License.
 
 package runtime
 
+// ObjectScheme represents common conversions between formal external API versions
+// and the internal Go structs. ObjectScheme is typically used with ObjectCodec to
+// transform internal Go structs into serialized versions. There may be many valid
+// ObjectCodecs for each ObjectScheme.
+type ObjectScheme interface {
+	ObjectConvertor
+	ObjectTyper
+	ObjectCreater
+	ObjectCopier
+}
+
+// ObjectCodec represents the common mechanisms for converting to and from a particular
+// binary representation of an object.
+type ObjectCodec interface {
+	ObjectEncoder
+	Decoder
```

Comment:
It does both sides (decode and encode) which we call a codec elsewhere - this is the more powerful interface (let's you pick output version).

> On May 1, 2015, at 12:06 PM, Daniel Smith <notifications@github.com> wrote:
> 
> In pkg/runtime/interfaces.go:
> 
> > +// ObjectScheme represents common conversions between formal external API versions
> > +// and the internal Go structs. ObjectScheme is typically used with ObjectCodec to
> > +// transform internal Go structs into serialized versions. There may be many valid
> > +// ObjectCodecs for each ObjectScheme.
> > +type ObjectScheme interface {
> > +	ObjectConvertor
> > +	ObjectTyper
> > +	ObjectCreater
> > +	ObjectCopier
> > +}
> > +
> > +// ObjectCodec represents the common mechanisms for converting to and from a particular
> > +// binary representation of an object.
> > +type ObjectCodec interface {
> > +	ObjectEncoder
> > +	Decoder
> ObjectDecoder?
> 
> —
> Reply to this email directly or view it on GitHub.
>